python/lsst/ip/diffim/subtractImages.py

```diff
@@ -772,26 +785,20 @@ def _sourceSelector(self, sources, mask):
             If there are too few sources to compute the PSF matching kernel
             remaining after source selection.
         """
-        flags = np.ones(len(sources), dtype=bool)
-        for flag in self.config.badSourceFlags:
-            try:
-                flags *= ~sources[flag]
-            except Exception as e:
-                self.log.warning("Could not apply source flag: %s", e)
-        signalToNoise = sources.getPsfInstFlux()/sources.getPsfInstFluxErr()
-        sToNFlag = signalToNoise > self.config.detectionThreshold
-        flags *= sToNFlag
-        sToNFlagMax = signalToNoise < self.config.detectionThresholdMax
-        flags *= sToNFlagMax
-        flags *= self._checkMask(mask, sources, self.config.excludeMaskPlanes)
-        selectSources = sources[flags].copy(deep=True)
+
+        selected = self.sourceSelector.selectSources(sources).selected
+        nInitialSelected = np.count_nonzero(selected)
+        selected *= self._checkMask(mask, sources, self.config.excludeMaskPlanes)
+        nSelected = np.count_nonzero(selected)
+        self.log.info("%i candidate sources rejected from template mask", nInitialSelected - nSelected)
+        selectSources = sources[selected].copy(deep=True)
         if (len(selectSources) > self.config.maxKernelSources) & (self.config.maxKernelSources > 0):
```

Comment:
Consider adding a one-line comment along the lines of `# Trim selectSources so it doesn't exceed maxKernalSources`. I presume `selectSources` is not ordered in any meaningful way? So we won't, for example, throw out all the sources in one region of the image, or those with certain S/N or brightness properties?